### DISCUSSION TITLE
I want to use custom aapt to rename apk resources

### DISCUSSION THREAD

**Discussion by Fablynn** [iBotPeaches/Apktool#3277] [Q&A] [answered]
i am a beginner, I want to use custom aapt to rename apk resources
I compiled aapt through Google's official tutorial, but found in the use of apktool that aapt only accepts names without $ symbols
So I started looking for a way to compile the author's aapt source code
I want to know what version of the Android source code can successfully compile the author's aapt

**Comment by iBotPeaches** [comment]
It used to be based off tags, but now it's just whatever was on top of master (main) when I pulled last. Probably not the best way by any stretch, but easier for sure on my end.

**Reply by Fablynn** [reply]
I have tried to directly replace the content of the decompiled resource file, the xml file and some image files can be directly replaced, but some file names are extended to the dex file
i don't know which pictures can change the name this way
But someone once used aapt to complete resource name obfuscation，By modifying makeFileResources() in Resource.cpp
I tried multiple versions of Google source code and found it difficult to successfully compile the aapt you provided in github
Now I have no idea how to solve this problem, can you help me, I will pay for it
I want to change the names of all resource files randomly when recompiling the apk
I've been struggling with this problem for five days

**Reply by iBotPeaches** [reply]
Sorry I struggle as-is building aapt, I wish you luck and hope my sources and documentation can help.

**Comment by Fablynn** [comment]
Thank you very much for your reply
I plan to pull an older version of master
May I ask you to compile aapt through the following steps
1.unbutu system build a compilation environment
2.The repo tool synchronizes the branch code
3.use the following command
make clobber
source build/envsetup.sh
lunch sdk-eng 
make -j9 aapt   

Here are my steps to compile
is it different from yours

**Reply by iBotPeaches** [reply, marked as answer]
Check my docs about building this - https://github.com/iBotPeaches/Apktool/blob/master/INTERNAL.md#building-aapt-binaries

**Reply by Fablynn** [reply]
Thank you for your help

**Reply by Fablynn** [reply]
I'm glad you provided documentation that made me feel at ease and ruled out a lot of bad ideas
Compile aapt according to your documentation and found an error
```error
# make aapt
14:53:09 Build sandboxing disabled due to nsjail error.
============================================
PLATFORM_VERSION_CODENAME=VanillaIceCream
PLATFORM_VERSION=VanillaIceCream
PRODUCT_INCLUDE_TAGS=com.android.mainline
TARGET_PRODUCT=sdk
TARGET_BUILD_VARIANT=eng
TARGET_ARCH=x86
TARGET_ARCH_VARIANT=x86
HOST_OS=linux
HOST_OS_EXTRA=Linux-3.10.0-1160.76.1.el7.x86_64-x86_64-Ubuntu-22.04.3-LTS
HOST_CROSS_OS=windows
BUILD_ID=AOSP.MAIN
OUT_DIR=out
============================================
[ 98% 437/445] converting Android.bp files to BUILD files at out/soong/bp2build
FAILED: out/soong/bp2build_files_marker
cd "$(dirname "out/host/linux-x86/bin/soong_build")" && BUILDER="$PWD/$(basename "out/host/linux-x86/bin/soong_build")" && cd / && env -i  "$BUILDER"     --top "$TOP"     --soong_out "out/soong"     --out "out"     --soong_variables out/soong/soong.sdk.variables --bp2build_marker out/soong/bp2build_files_marker --globListDir bp2build_files --globFile out/soong/globs-bp2build_files.ninja -t -l out/.module_paths/Android.bp.list --available_env out/soong/soong.environment.available --used_env out/soong/soong.environment.used.sdk.bp2build_files Android.bp
error: frameworks/base/tools/split-select/Android.bp:60:12: unrecognized property "product_variables.pdk"
fatal errors encountered
14:54:40 soong bootstrap failed with: exit status 1

#### failed to build some targets (01:33 (mm:ss)) ####
```
Maybe solve this problem and I can successfully compile aapt

**Reply by iBotPeaches** [reply]
When Android 14 drops, I'll update that repo with bleeding edge at that moment of main.

**Reply by Fablynn** [reply]
It's great, I'm looking forward to it

**Reply by Fablynn** [reply]
I will continue to study this problem, I hope this problem is not particularly difficult
